src/controllers/hibp.js

```diff
@@ -0,0 +1,110 @@
+import AppConstants from '../app-constants.js'
+import { getBreachByName, loadBreachesIntoApp } from '../utils/hibp.js'
+import mozlog from '../utils/log.js'
+const log = mozlog('controllers.hibp')
+
+/**
+ * Whenever a breach is detected on the HIBP side, HIBP sends a request to this endpoint.
+ * This function attempts to retrieve the breach info from the local cache, if not found
+ * it retrieves it from the database
+ * A breach notification contains the following parameters:
+ * - breachName
+ * - hashPrefix
+ * - hashSuffixes
+ * More about how account identities are anonymized: https://blog.mozilla.org/security/2018/06/25/scanning-breached-accounts-k-anonymity/
+ */
+async function notify (req, res) {
+  if (!req.token || req.token !== AppConstants.HIBP_NOTIFY_TOKEN) {
+    const errorMessage = 'HIBP notify endpoint requires valid authorization token.'
+    throw new Error(errorMessage)
+  }
+  if (!['breachName', 'hashPrefix', 'hashSuffixes'].every(req.body?.hasOwnProperty, req.body)) {
+    throw new Error('Breach notification requires breachName, hashPrefix, and hashSuffixes.')
+  }
+
+  const reqBreachName = req.body.breachName.toLowerCase()
```

Comment:
We can't be sure, since it's up to the requester. We do know that we only support all lowercase

Reply:
I meant our function -- aren't we already changing to lowercase in utils/hibp.js ?
https://github.com/mozilla/blurts-server/blob/4fb6c993a7c573c1d0589b85e4b9808d34e78d07/src/utils/hibp.js#L193-L200

Reply:
Ah yes. Thanks for pointing that one. 
I've changed it, though one minor issue is that now it's possible that when we throw the error on line 37, it will still show the requested breach name without lowercase
```
throw new Error('Unrecognized breach: ' + breachName)
```